Modules/Sources/Yosemite/Stores/BookingStore.swift

```diff
@@ -263,9 +263,39 @@ private extension BookingStore {
             siteID: siteID,
             bookingID: bookingID,
             statusKey: status
-        ) { _ in
-            //TODO: - booking status remote update + rollback status in case of error
-            onCompletion(nil)
+        ) { [weak self] previousStatusKey in
+            guard let self else {
+                return onCompletion(UpdateBookingStatusError.undefinedState)
+            }
+
+            Task {
```

Comment:
❓ Is it necessary to trigger the completion closure on the main thread with `@MainActor` here?